internal/tools/httpjson/httpjson.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.package http
+package httpjson
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"maps"
+
+	"github.com/googleapis/genai-toolbox/internal/sources"
+	httpsrc "github.com/googleapis/genai-toolbox/internal/sources/http"
+	"github.com/googleapis/genai-toolbox/internal/tools"
+)
+
+const ToolKind string = "http-json"
+
+type responseBody []byte
+
+type Config struct {
+	Name         string            `yaml:"name" validate:"required"`
+	Kind         string            `yaml:"kind" validate:"required"`
+	Source       string            `yaml:"source" validate:"required"`
+	Description  string            `yaml:"description" validate:"required"`
+	AuthRequired []string          `yaml:"authRequired"`
+	Path         string            `yaml:"path" validate:"required"`
+	Method       tools.HTTPMethod  `yaml:"method" validate:"required"`
+	Headers      map[string]string `yaml:"headers"`
+	RequestBody  string            `yaml:"requestBody"`
+	QueryParams  tools.Parameters  `yaml:"queryParams"`
+	BodyParams   tools.Parameters  `yaml:"bodyParams"`
+	HeaderParams tools.Parameters  `yaml:"headerParams"`
+}
+
+// validate interface
+var _ tools.ToolConfig = Config{}
+
+func (cfg Config) ToolConfigKind() string {
+	return ToolKind
+}
+
+func (cfg Config) Initialize(srcs map[string]sources.Source) (tools.Tool, error) {
+	// verify source exists
+	rawS, ok := srcs[cfg.Source]
+	if !ok {
+		return nil, fmt.Errorf("no source named %q configured", cfg.Source)
+	}
+
+	// verify the source is compatible
+	s, ok := rawS.(*httpsrc.Source)
+	if !ok {
+		return nil, fmt.Errorf("invalid source for %q tool: source kind must be `http`", ToolKind)
+	}
+
+	// Create URL based on BaseURL and Path
+	// Attach query parameters
+	u, err := url.Parse(s.BaseURL + cfg.Path)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing URL: %s", err)
+	}
+
+	queryParameters := u.Query()
```

Comment:
nit: add a coment -- get any existing query parameters from the URL so they are preserved